source/common/network/listen_socket_impl.h

```diff
@@ -74,12 +75,19 @@ template <typename T> class NetworkListenSocket : public ListenSocketImpl {
 
   NetworkListenSocket(IoHandlePtr&& io_handle, const Address::InstanceConstSharedPtr& address,
                       const Network::Socket::OptionsSharedPtr& options)
-      : ListenSocketImpl(std::move(io_handle), address) {
+      : ListenSocketImpl(std::move(io_handle), address), bind_to_port_(io_handle_ == nullptr) {
```

Comment:
I don't really understand this code. if io_handle_ is nullptr, why do we need all these overrides? Can't we just check that directly in the various functions?

Reply:
The background is that UDP bind_to_port_ has different meaning. 
The default override which delegates to `ListenSocketImpl` is for UDP. It could be some guard code at udp.

Let me run a full tests to see if UDP does the same logic on bind_to_port to see if there is any break.

Reply:
FIxed!